WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -1094,16 +1095,18 @@ public void initSelectedSite() {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onPostUploaded(OnPostUploaded event) {
-        SiteModel site = getSelectedSite();
-        if (site != null && event.post != null && event.post.getLocalSiteId() == site.getId()) {
-            UploadUtils.onPostUploadedSnackbarHandler(
-                    this,
-                    findViewById(R.id.coordinator),
-                    event.isError(),
-                    event.post,
-                    null,
-                    site,
-                    mDispatcher);
+        if (getLifecycle().getCurrentState().isAtLeast(STARTED)) {
```

Comment:
We may want to add a comment for why we need to check for this, but it's fairly obvious so I am not sure. Feel free to make this addition if you'd like as a part of any other PR. Thanks!